compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1943,12 +1943,26 @@ object desugar {
       case AndType(tp1, tp2) => stripToCore(tp1) ::: stripToCore(tp2)
       case _ => defn.AnyType :: Nil
     }
+
+    val refinements1 = refinements.flatMap {
```

Comment:
It's more efficient to make this a `refinements.mapConserve(...).flatten`. If there are no `var` refinements this would never copy the trees.